src/net/sourceforge/kolmafia/Modeable.java

```diff
@@ -139,18 +149,20 @@ public static Modeable find(final String command) {
   }
 
   public static Modeable find(final int itemId) {
-    return Arrays.stream(values())
-        .filter(m -> m.getItem().getItemId() == itemId)
-        .findAny()
-        .orElse(null);
+    return idToModeable.get(itemId);
   }
 
   public static Map<Modeable, String> getStringMap(Function<Modeable, String> cb) {
     return Arrays.stream(values()).collect(Collectors.toMap(m -> m, cb));
   }
 
   public static Map<Modeable, String> getStateMap() {
-    return getStringMap(Modeable::getState);
+    // This is performance-critical for the maximizer and needs to be written imperatively.
```

Comment:
It is a real shame just how non-performant streams are when compared with for loops. I miss languages like C, where the compiler is a magic wand that can detect code that looks like it's doing a certain operation and optimise it for you.

Reply:
Agreed, sadly. It's not horrendously slower but enough to be noticeable, and this gets called very frequently.